2.6/root/usr/share/container-scripts/mongodb/replset-supervisor.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# This script registers the current container into MongoDB replica set or initiate replSet if $1 == "initiate"
+
+source ${CONTAINER_SCRIPTS_PATH}/base-functions.sh
+source ${CONTAINER_SCRIPTS_PATH}/init-functions.sh
+source ${CONTAINER_SCRIPTS_PATH}/replset-functions.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+set -x
+
+wait_mongo "UP"
+
+if [[ "$1" == "initiate" ]]; then
+  # Prepare master
+  echo "=> Initiating the replSet ${MONGODB_REPLICA_NAME} ..."
+
+  mongo_initiate
+
+  echo "=> Creating MongoDB users ..."
+  mongo_create_admin
+  mongo_create_user "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+
+  # Add one pod to this replSet to be able to address it with mongodb service
+  member_addr=$(echo "$(endpoints)" | head -n 1)
+  echo "=> Adding first member ${member_addr} to replSet ..."
+  mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --eval "rs.add('${member_addr}:27017');"
+
+  mongo_wait_replset "-u admin -p ${MONGODB_ADMIN_PASSWORD}" "$(mongo_addr)"
+
+  echo "=> Successfully initialized replSet ..."
+
+  # Wait for at least 3 members
+  members=0
+  while [ "${members}" -le 3 ] 2>/dev/null || [[ $? -ne 1 ]] ; do
+    echo "=> Waiting for other replSet members ..."
+    sleep ${SLEEP_TIME}
+    mongo_wait_replset "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+    members=$(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --quiet --host "$(replset_addr)" --eval "rs.status().members.length" | tail -n 1)
+  done
+
+  # Exit this pod
+  kill $2
+
+else
+  echo "=> Trying to add this mongod to the replSet ..."
+  # Add to the replSet
+  mongo_wait_replset "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+
+  # Add the current container to the replSet
+  mongo_add
+
+  echo "=> Successfully added to the replSet ..."
+
+fi
```

Comment:
> I don't know what @php-coder had in mind, but for me functions are more than being able to call them in multiple places, they can give a name to a procedure and make reading and understanding the code easier.

The code is separated in a separate file (to note - it same to container-scripts/mongodb/initiate_replica.sh file - should everything be "separated" in a new function?).

IMPOV a separation of almost all code in a file to make a function from it to "hide" IF-ELSE statement to contains only one line in each branch with function call is not the way to implement "coding for others to read" :-(

However if this is a style in OpenShift I can do it.